java/core/src/main/java/com/redhat/rhn/manager/setup/ProductSyncManager.java

```diff
@@ -451,6 +450,15 @@ private SetupWizardProductDto convertProduct(final MgrSyncProductDto productIn)
         return displayProduct;
     }
 
+    private static int sortBaseChannelOnTop(MgrSyncChannelDto a, MgrSyncChannelDto b) {
+        if (StringUtils.isBlank(a.getParentLabel())) {
+            return -1;
+        }
+        else {
+            return StringUtils.isBlank(b.getParentLabel()) ? 1 : 0;
+        }
+    }
```

Comment:
Not due to your change, but this comparator has a bug. If both `a` and `b` have blank parent label, both `sortBaseChannelOnTop(a, b)` and `sortBaseChannelOnTop(b, a)`  returns `-1`. Moreover, `sortBaseChannelOnTop(a, a)` also returns `-1' instead of `0`'.  So this code breaks the compare contract. I think this should simply be replaced by:

```java
        List<Channel> mandatoryChannelsOut = collect.get(true).stream()
            // Add base channel on top of everything else so it can be added first.
            // Base channels have empty parent, and java sorts false before true
            .sorted(Comparator.comparing(dto -> StringUtils.isNotBlank(dto.getParentLabel())))
            .map(mapping)
            .collect(Collectors.toList());
```